ERCS/erc-8006.md

```diff
@@ -0,0 +1,370 @@
+---
+eip: 8006
+title: Universal Policy Engine
+description: Composite on-chain policy engine utilizing generic rule artifacts within DAG evaluator
+author: Vlad Pryadko (@VladPryad) <vladyslav.priad@gmail.com>, Vitali Grabovski (@vitaligrabovski1) <v.grabovski@guardianlabs.org>
+discussions-to: https://ethereum-magicians.org/t/erc-8006-universal-policy-engine/25155
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-03-14
+---
+
+## Abstract
+
+Smart contract applications often need configurable and reusable compliance logic. This standard defines an on-chain policy engine that composes rule "artifacts" and evaluates the final check result based on them through a specialized rule processor called the Policy Handler.
+
+A **Directed Acyclic Graph (DAG)** is the RECOMMENDED approach for composition and evaluation in Policy Handler. Artifacts implement a minimal interface and exchange input params / output params of the main interface methods as `bytes` or `bytes[]` (see [Interfaces section](#interfaces)), allowing heterogeneous data and on/off-chain variable supply.
```

Comment:
Please keep requirements (indicated by UPPERCASE keywords) in the Specification section.